pkg/plugin/driver.go

```diff
@@ -227,8 +230,15 @@ func (h *Clickhouse) MutateQuery(ctx context.Context, req backend.DataQuery) (co
 		return ctx, req
 	}
 
+	customSettings := make(clickhouse.Settings)
+	if dataQuery.CustomSettings != nil {
+		for _, setting := range dataQuery.CustomSettings {
+			customSettings[setting.Setting] = setting.Value
+		}
+	}
+
 	loc, _ := time.LoadLocation(dataQuery.Meta.TimeZone)
-	return clickhouse.Context(ctx, clickhouse.WithUserLocation(loc)), req
+	return clickhouse.Context(ctx, clickhouse.WithUserLocation(loc), clickhouse.WithSettings(customSettings)), req
```

Comment:
@bossinc In this function `MutateQuery`, is there a way to access:

- Datasource settings (`backend.DataSourceInstanceSettings`)?
- Currently logged-in user to Grafana?

If so, I could remove client-side passing/templating of settings and do this entirely in the backend.